### PR TITLE
[WIP] ENH: Add support for building interfaces from Boutiques specs

### PR DESCRIPTION
**Pull request by effigies**
## Summary

Initial support for dynamically constructing Nipype interfaces from Boutiques specifications

## List of changes proposed in this PR (pull-request)

* [x] Create `BoutiquesInterface` class with dynamic input/output specs
* [x] Populate the input spec based on the `"inputs"` section of a passed spec
* [x] Populate the output spec based on the `"output-files"` section of a passed spec
* [ ] Test submission in `MultiProc`, to ensure proper serialization/deserialization
* [ ] Add basic AFNI test (considering [adwarp](https://github.com/afni/afni_boutiques/blob/master/adwarp.json) as it is relatively complete)
* [ ] Add script, or small set of scripts, with specs that exercise all features of the schema

cc @gkiar @leej3 

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by effigies** [comment]
I'm somewhat inclined to target Nipype 2.0 for this, but I don't know that there's anything that will be made particularly easier by dropping Python 2 support.

File: nipype/interfaces/base/boutiques.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    input_spec = DynamicTraitedSpec
+    output_spec = DynamicTraitedSpec
+
+    # Subclasses may override the trait_map to provide better traits
+    # for various types
+    trait_map = {
+        'File': File,
+        'String': Str,
+        'Number': traits.Float,
```
Comment (satra) [inline nipype/interfaces/base/boutiques.py]: @gkiar - is it really only `Number`? shouldn't boutiques support at least the XSD literal datatypes?

**Comment by satra** [comment]
thanks @effigies - targeting 2.0 is fine with me. this will be a good sprint item to finalize, together with CWL support.

**Comment by effigies** [comment]
Just as a note, most of the reason for creating the `trait_map` class variable was so that an `AFNIBoutiqueInterface` could adapt `File` to handle AFNI's command line specs.

File: nipype/interfaces/base/boutiques.py (hunk 1 above)
Comment (effigies) [reply nipype/interfaces/base/boutiques.py]: There is a separate "integer" constraint, as well.

**Comment by gkiar** [comment]
cc: @glatard

**Pull request closed without merging**